lib/valueflow.cpp

```diff
@@ -1692,16 +1705,115 @@ static bool valueFlowForward(Token * const               startToken,
         const Scope * scope = symboldatabase->functionScopes[i];
-        for (Token* tok = const_cast<Token*>(scope->classStart); tok != scope->classEnd; tok = tok->next()) {
-            // Assignment
-            if ((tok->str() != "=") || (tok->astParent()))
+        if (!scope)
+            continue;
+        const Token * start = scope->classStart;
+        if (scope->function) {
+            const Token * memberInitializationTok = scope->function->constructorMemberInitialization();
+            if (memberInitializationTok)
+                start = memberInitializationTok;
+        }
+
+        for (Token* tok = const_cast<Token*>(start); tok != scope->classEnd; tok = tok->next()) {
+            Token * varTok;
+            ValueFlow::Value::MoveKind moveKind;
+            if (!isStdMoveOrStdForwarded(tok, &moveKind, &varTok))
                 continue;
+            // x is not MOVED after assignment if code is:  x = ... std::move(x) .. ;
+            const Token *parent = tok->astParent();
+            while (parent && parent->str() != "=" && parent->str() != "return")
+                parent = parent->astParent();
+            if (parent && parent->str() == "return") // MOVED in return statement
+                continue;
+            if (parent && parent->astOperand1()->str() == varTok->str())
+                continue;
+            const unsigned int varid = varTok->varId();
+            const Variable *var = varTok->variable();
+            if (!var)
+                continue;
+            const Token * const endOfVarScope = var->typeStartToken()->scope()->classEnd;
+
+            ValueFlow::Value value;
+            value.valueType = ValueFlow::Value::MOVED;
+            value.moveKind = moveKind;
+            value.setKnown();
+            std::list<ValueFlow::Value> values;
+            values.push_back(value);
+
+            valueFlowForward(varTok->next(), endOfVarScope, var, varid, values, false, tokenlist, errorLogger, settings);
+        }
+    }
+}
+
+static const Token * nextAfterAstRightmostLeaf(Token const * tok)
+{
+    const Token * rightmostLeaf = tok;
+    if (!rightmostLeaf || !rightmostLeaf->astOperand1())
+        return nullptr;
+    do {
+        if (rightmostLeaf->astOperand2())
+            rightmostLeaf = rightmostLeaf->astOperand2();
+        else
+            rightmostLeaf = rightmostLeaf->astOperand1();
+    } while (rightmostLeaf->astOperand1());
+    return rightmostLeaf->next();
+}
+
+static void valueFlowAfterAssign(TokenList *tokenlist, SymbolDatabase* symboldatabase, ErrorLogger *errorLogger, const Settings *settings)
+{
+    const std::size_t functions = symboldatabase->functionScopes.size();
+    for (std::size_t i = 0; i < functions; ++i) {
+        const Scope * scope = symboldatabase->functionScopes[i];
+        const Token * start = scope->classStart;
+        if (scope->function) {
+            const Token * memberInitializationTok = scope->function->constructorMemberInitialization();
```

Comment:
as far as I see this is redundant. The valueFlowAfterAssign() will not handle the initializer list as far as I see.